### PR TITLE
fix: shorthand camelize should be used in `rawName`

### PR DESCRIPTION
**Pull request by waynzh**
This PR addressed the issue reported in https://github.com/vuejs/eslint-plugin-vue/issues/2370.

I believe the issue is from `vue-eslint-parser` transforming the `name` into camel case. For example, the `name` for `:fooBar` is being interpreted as `foobar`, which won't match with the variable `fooBar`.

This PR proposes camelizing the `rawName` instead, which should hopefully resolve the problem.

I'll throw in some test cases soon if this fix makes sense.

### CODE REVIEW

**Comment by ota-meshi** [comment]
Thank you for this PR!
Could you please add a test case (fixture) where we can check that we was able to fix it?

**Review by ota-meshi — CHANGES_REQUESTED** [review]
Thank you for this PR!
Could you please add a test case (fixture) where we can check that we was able to fix it?

File: test/fixtures/ast/v-bind-same-name-shorthand07-camelcase/ast.json [hunk 1]
```diff
@@ -0,0 +1,968 @@
...
+                                        "end": {
+                                            "column": 14,
+                                            "line": 6
+                                        }
+                                    },
+                                    "range": [
+                                        91,
+                                        97
+                                    ],
+                                    "name": "srcUrl"
```
Comment (waynzh) [inline test/fixtures/ast/v-bind-same-name-shorthand07-camelcase/ast.json:423]: The main change here is modifying the original `srcurl` to `srcUrl`.

File: test/fixtures/ast/v-bind-same-name-shorthand07-camelcase/scope.json [hunk 2]
```diff
@@ -0,0 +1,154 @@
+{
+    "type": "global",
+    "variables": [],
+    "references": [],
+    "childScopes": [
+        {
+            "type": "module",
+            "variables": [
+                {
+                    "name": "src",
```
Comment (ota-meshi) [inline test/fixtures/ast/v-bind-same-name-shorthand07-camelcase/scope.json:10]: Could you please run `npm run update-fixtures`?

**Review by ota-meshi — CHANGES_REQUESTED** [review]

File: test/fixtures/ast/v-bind-same-name-shorthand07-camelcase/scope.json (hunk 2 above)
Comment (waynzh) [reply test/fixtures/ast/v-bind-same-name-shorthand07-camelcase/scope.json]: done ✅

**Review by ota-meshi — APPROVED** [review]
LGTM! Thank you!

**Pull request merged**